quil-rs/src/program/mod.rs

```diff
@@ -74,6 +74,8 @@ pub struct Program {
     pub memory_regions: BTreeMap<String, MemoryRegion>,
     pub waveforms: BTreeMap<String, Waveform>,
     instructions: Vec<Instruction>,
+    // private field used for caching operations
+    used_qubits: HashSet<Qubit>,
```

Comment:
```suggestion
    /// The qubits used by all of the instructions; that is, the qubits on which the program operates.
    /// Note that this does not include all qubits _referenced_ by the program (such as those within frames defined but never used)
    used_qubits: HashSet<Qubit>,
```

The other fields may not be `///` documented, but this one can be the first 😄